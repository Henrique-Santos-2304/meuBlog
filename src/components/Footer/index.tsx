import { Text, Container } from "@chakra-ui/react";

const Footer = () => (
  <Container
    maxW="full"
    height="4vh"
    bgGradient="linear(to-l,#00bf72 ,#008793 ,#004d7a , #051937)"
  >
    <Container>
      <Text
        color="gray.100"
        isTruncated
        textAlign="end"
        fontSize="2xl"
        maxW="container.md"
      >
        Criado por <b style={{ color: "yellow" }}>Henrique dos Santos</b>
      </Text>
    </Container>
  </Container>
);

export default Footer;
