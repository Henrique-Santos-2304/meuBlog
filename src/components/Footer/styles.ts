import styled, { css } from "styled-components";
export const Wrapper = styled.div`
  margin: 0 auto;
  width: ${({ theme }) => theme.grid.containerContent};
  height: 8rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0.2rem 3rem;
`;
export const WrapperContent = styled.div`
  ${({ theme }) => css`
    width: ${theme.grid.container};
    position: fixed;
    bottom: 0;
  `}
`;
export const WrapperSocials = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    color: ${theme.colors.white};
    font-size: ${theme.font.sizes.xbiglarge};
    margin-left: 8rem;
  `}
`;
export const IconUp = styled.a`
  ${({ theme }) => css`
    color: ${theme.colors.white};
    transform: rotate(-90deg);
    margin-right: 8rem;
    cursor: pointer;

    svg {
      font-size: 6rem;
    }
  `}
`;
