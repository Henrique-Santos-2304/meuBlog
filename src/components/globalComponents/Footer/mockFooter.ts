export const mockFooter = {
  dataFooter: {
    ariaLabel: "links para contatos e portfolio",
    iconsContactSocial: [
      {
        url: "http://github.com",
        titleIcon: "/meugithub",
        ariaLabel: "github",
      },
      {
        url: "http://linkedin.com",
        titleIcon: "/meulinkedin",
        ariaLabel: "linkedin",
      },
      {
        url: "http://whatsapp.com",
        titleIcon: "/meuwhatsapp",
        ariaLabel: "whatsapp",
      },
    ],
  },
};
