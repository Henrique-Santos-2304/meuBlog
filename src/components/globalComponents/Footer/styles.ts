import styled from "styled-components";
export const ContainerFooter = styled.footer`
  width: 100vw;
  margin: 0 auto;
  background: rgba(0, 0, 0, 0.55);
  clip-path: polygon(0 0, 100% 15%, 100% 100%, 0 100%);
`;
