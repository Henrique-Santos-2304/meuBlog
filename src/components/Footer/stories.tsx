import { Story, Meta } from "@storybook/react";
import Footer from ".";

export default {
  title: "Footer",
  component: Footer,
} as Meta;

export const Basic: Story = (args) => <Footer {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Footer {...args} />;
