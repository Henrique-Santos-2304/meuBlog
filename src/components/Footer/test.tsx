import Footer from ".";
import { renderTheme } from "utils/testRenderTheme";

describe("<Footer />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
