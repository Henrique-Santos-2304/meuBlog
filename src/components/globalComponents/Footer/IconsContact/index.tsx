import * as S from "./styles";
import { iconSocial } from "graphql/typesQueries/types";

type propsIconsSocials = {
  iconsContactSocial: iconSocial[];
};
const IconsContact = ({ iconsContactSocial }: propsIconsSocials) => (
  <>
    <S.Wrapper>
      {iconsContactSocial.map((icon) => (
        <S.Links href={icon.url} passHref key={icon.titleIcon}>
          <S.Anchor title={icon.titleIcon}>
            <S.ContentIcon>
              {icon.ariaLabel === "github" && <S.Github />}
              {icon.ariaLabel === "linkedin" && <S.Linkedin />}
              {icon.ariaLabel === "whatsapp" && <S.Whatsapp />}
            </S.ContentIcon>
          </S.Anchor>
        </S.Links>
      ))}
    </S.Wrapper>
  </>
);

export default IconsContact;
