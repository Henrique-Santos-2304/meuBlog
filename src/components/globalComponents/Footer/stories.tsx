import { Story, Meta } from "@storybook/react";
import Footer from ".";
import { mockFooter } from "./mockFooter";

export default {
  title: "Footer",
  component: Footer,
} as Meta;

export const Basic: Story = () => <Footer {...mockFooter} />;

export const Default: Story = () => <Footer {...mockFooter} />;
