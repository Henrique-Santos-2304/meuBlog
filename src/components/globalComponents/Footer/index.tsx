import { footerIcon } from "graphql/typesQueries/types";
import IconsContact from "./IconsContact";
import * as S from "./styles";

type propsFooter = {
  dataFooter: footerIcon;
};
const Footer = ({ dataFooter }: propsFooter) => (
  <S.ContainerFooter aria-label={dataFooter.ariaLabel}>
    <IconsContact iconsContactSocial={dataFooter.iconsContactSocial} />
  </S.ContainerFooter>
);

export default Footer;
