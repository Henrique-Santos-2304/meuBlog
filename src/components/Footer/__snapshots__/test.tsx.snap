// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render  1`] = `
<div>
   
  <div
    class="chakra-container css-gnbstb"
  >
    <div
      class="chakra-container css-10gf8jm"
    >
      <p
        class="chakra-text css-1bg3qoe"
      >
        Criado por 
        <b
          style="color: yellow;"
        >
          Henrique dos Santos
        </b>
      </p>
    </div>
  </div>
</div>
`;
